### PR TITLE
Batch states now include running, complete, and failed

### LINKED ISSUE #710: Set of batch states != set of job states
Two different enums :pensive: 

Renaming batch `launched` to `running` would solve it.

Arguably, batches could also get `complete` and `failed`, but... eh.

### PR DESCRIPTION
**Pull request by hkethi002**
Fixes #710 

Batch is complete when all jobs are in a final (stable) state and is failed if all jobs are in the failed state.
### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

File: api/jobs/batch.py [hunk 1]
```diff
@@ -161,7 +163,8 @@ def run(batch_job):
                 'config':   config_,
                 'gear_id':  gear_id,
                 'inputs':   inputs,
-                'tags':     tags
+                'tags':     tags,
+                'batch':    "batch is" + str(batch_job.get('_id'))
```
Comment (nagem) [inline api/jobs/batch.py]: Looks like some testing code made it's way into the PR.

File: test/integration_tests/python/test_batch.py [hunk 2]
```diff
@@ -126,3 +128,59 @@ def test_batch(data_builder, as_user, as_admin, as_root):
...
+        r = as_root.put('/jobs/' + job, json={'state': 'running'})
+        r = as_root.put('/jobs/' + job, json={'state': 'complete'})
+        assert r.ok
+
+    # test batch is complete
+    r = as_admin.get('/batch/' + batch_id)
+    assert r.json()['state'] == 'complete'
+
+    # Test batch failed
+    # create a batch w/ acquisition target and target_context
```
Comment (nagem) [inline test/integration_tests/python/test_batch.py]: Could we also add a test for an analysis target?

File: api/jobs/batch.py [hunk 3]
```diff
@@ -206,6 +209,25 @@ def cancel(batch_job):
     update(batch_job['_id'], {'state': 'cancelled'})
     return cancelled_jobs
 
+def check_state(batch_id):
+    """
+    Return True if all jobs are complete or failed
+    """    
+
+    batch = config.db.batch.find_one({'_id': batch_id})
```
Comment (nagem) [inline api/jobs/batch.py]: Mind using the local get method instead of accessing the db directly?

File: api/jobs/batch.py [hunk 4]
```diff
@@ -206,6 +209,25 @@ def cancel(batch_job):
     update(batch_job['_id'], {'state': 'cancelled'})
     return cancelled_jobs
 
+def check_state(batch_id):
+    """
+    Return True if all jobs are complete or failed
```
Comment (nagem) [inline api/jobs/batch.py]: The description mentions it's a True/False response but it looks like it returns a state below.

File: api/jobs/batch.py [hunk 5]
```diff
@@ -206,6 +209,25 @@ def cancel(batch_job):
...
+    """    
+
+    batch = config.db.batch.find_one({'_id': batch_id})
+
+    if batch.get('state') == 'cancelled':
+        return False
```
Comment (nagem) [inline api/jobs/batch.py]: Maybe return "None" here, it makes more sense with the string responses.

**Comment by kofalt** [comment]
Presumably this will need a DB upgrade, yeah?

**Comment by nagem** [comment]
Oh yeah, nice catch @kofalt.

@hkethi002 we could import the function you use to determine status and just iterate over the batch items.

**Comment by nagem** [comment]
Changes look good.

File: api/jobs/queue.py [hunk 6]
```diff
@@ -72,7 +72,6 @@ def mutate(job, mutation):
         if result.modified_count != 1:
             raise Exception('Job modification not saved')
 
-        # If the job did not succeed, check to see if job should be retried.
```
Comment (kofalt) [inline api/jobs/queue.py]: Why was this comment removed?

Comment (hkethi002) [inline api/jobs/queue.py]: That was probably removed by mistake, I'll that back in, thanks for catching it!

**Comment by kofalt** [comment]
Requires SDK signoff

**Pull request merged**